scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala

```diff
@@ -534,7 +534,7 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
           case (_: RegionDelim) :: (_: RegionFor) :: _ => sepRegions
           case xs => RegionIf(next) :: xs
         })
-      case _: KwThen if dialect.allowSignificantIndentation =>
+      case _: KwThen =>
```

Comment:
Why the change with `then` ? Could you add a test for that?

Reply:
no test. i mentioned in the commit comment that `then` only exists in scala3, so the dialect check is probably redundant, just like with `do`.